File: Nx3CS/bin/test/src/nx3/NHead.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class NHead : global::haxe.lang.HxObject {
		public    NHead(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    NHead(global::nx3.EHeadType type, global::haxe.lang.Null<int> level, global::nx3.ESign sign, global::nx3.ETie tie, global::nx3.ETie tieTo){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\NHead.hx"
				global::nx3.NHead.__hx_ctor_nx3_NHead(this, type, level, sign, tie, tieTo);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_NHead(global::nx3.NHead __temp_me69, global::nx3.EHeadType type, global::haxe.lang.Null<int> level, global::nx3.ESign sign, global::nx3.ETie tie, global::nx3.ETie tieTo){
			unchecked {
				#line 22 "F:\\nx3\\src\\nx3\\NHead.hx"
				global::haxe.lang.Null<int> __temp_level68 = ( ( ! (level.hasValue) ) ? (new global::haxe.lang.Null<int>(0, true)) : (((global::haxe.lang.Null<int>) (level) )) );
				if (( type != default(global::nx3.EHeadType) )) {
					#line 23 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.type = type;
				}
				 else {
					#line 23 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.type = global::nx3.EHeadType.Normal;
				}
				
				#line 24 "F:\\nx3\\src\\nx3\\NHead.hx"
				if (( sign != default(global::nx3.ESign) )) {
					#line 24 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.sign = sign;
				}
				 else {
					#line 24 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.sign = global::nx3.ESign.None;
				}
				
				#line 25 "F:\\nx3\\src\\nx3\\NHead.hx"
				if (( tie != default(global::nx3.ETie) )) {
					#line 25 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.tie = tie;
				}
				 else {
					#line 25 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.tie = default(global::nx3.ETie);
				}
				
				#line 26 "F:\\nx3\\src\\nx3\\NHead.hx"
				if (( tieTo != default(global::nx3.ETie) )) {
					#line 26 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.tieTo = tieTo;
				}
				 else {
					#line 26 "F:\\nx3\\src\\nx3\\NHead.hx"
					__temp_me69.tieTo = default(global::nx3.ETie);
				}
				
				#line 27 "F:\\nx3\\src\\nx3\\NHead.hx"
				__temp_me69.level = __temp_level68.@value;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				return new global::nx3.NHead(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				return new global::nx3.NHead(((global::nx3.EHeadType) (arr[0]) ), global::haxe.lang.Null<object>.ofDynamic<int>(arr[1]), ((global::nx3.ESign) (arr[2]) ), ((global::nx3.ETie) (arr[3]) ), ((global::nx3.ETie) (arr[4]) ));
			}
			#line default
		}
		
		
		public  int level;
		
		public  global::nx3.EHeadType type;
		
		public  global::nx3.ESign sign;
		
		public  global::nx3.ETie tie;
		
		public  global::nx3.ETie tieTo;
		
		public  global::nx3.NNote nnote;
		
		public virtual   string toString(){
			unchecked {
				#line 32 "F:\\nx3\\src\\nx3\\NHead.hx"
				string str = global::haxe.lang.Runtime.concat("", global::haxe.lang.Runtime.toString(this.level));
				if (( this.type != global::nx3.EHeadType.Normal )) {
					#line 33 "F:\\nx3\\src\\nx3\\NHead.hx"
					str = global::haxe.lang.Runtime.concat(str, global::haxe.lang.Runtime.concat(" ", global::Type.enumConstructor(this.type)));
				}
				 else {
					#line 33 "F:\\nx3\\src\\nx3\\NHead.hx"
					str = global::haxe.lang.Runtime.concat(str, "");
				}
				
				#line 34 "F:\\nx3\\src\\nx3\\NHead.hx"
				if (( this.sign != global::nx3.ESign.None )) {
					#line 34 "F:\\nx3\\src\\nx3\\NHead.hx"
					str = global::haxe.lang.Runtime.concat(str, global::haxe.lang.Runtime.concat(" ", global::Type.enumConstructor(this.sign)));
				}
				 else {
					#line 34 "F:\\nx3\\src\\nx3\\NHead.hx"
					str = global::haxe.lang.Runtime.concat(str, "");
				}
				
				#line 35 "F:\\nx3\\src\\nx3\\NHead.hx"
				return global::haxe.lang.Runtime.concat(global::haxe.lang.Runtime.concat("NHead(", str), ")");
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				switch (hash){
					case 1919096196:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.level = ((int) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				switch (hash){
					case 522053472:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.nnote = ((global::nx3.NNote) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					case 271556331:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.tieTo = ((global::nx3.ETie) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					case 5792080:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.tie = ((global::nx3.ETie) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					case 1280544829:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.sign = ((global::nx3.ESign) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					case 1292432058:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.type = ((global::nx3.EHeadType) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					case 1919096196:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						this.level = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				switch (hash){
					case 946786476:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("toString") ), ((int) (946786476) ))) );
					}
					
					
					case 522053472:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.nnote;
					}
					
					
					case 271556331:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.tieTo;
					}
					
					
					case 5792080:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.tie;
					}
					
					
					case 1280544829:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.sign;
					}
					
					
					case 1292432058:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.type;
					}
					
					
					case 1919096196:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.level;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				switch (hash){
					case 1919096196:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return ((double) (this.level) );
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				switch (hash){
					case 946786476:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return this.toString();
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("nnote");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("tieTo");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("tie");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("sign");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("type");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				baseArr.push("level");
				#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
				{
					#line 11 "F:\\nx3\\src\\nx3\\NHead.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
		public override string ToString(){
			return this.toString();
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/NHead.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class NHead extends haxe.lang.HxObject
{
	public    NHead(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    NHead(nx3.EHeadType type, java.lang.Object level, nx3.ESign sign, nx3.ETie tie, nx3.ETie tieTo)
	{
		nx3.NHead.__hx_ctor_nx3_NHead(this, type, level, sign, tie, tieTo);
	}
	
	
	public static   void __hx_ctor_nx3_NHead(nx3.NHead __temp_me62, nx3.EHeadType type, java.lang.Object level, nx3.ESign sign, nx3.ETie tie, nx3.ETie tieTo)
	{
		java.lang.Object __temp_level61 = ( (( level == null )) ? (((java.lang.Object) (0) )) : (((java.lang.Object) (level) )) );
		if (( type != null )) 
		{
			__temp_me62.type = type;
		}
		 else 
		{
			__temp_me62.type = nx3.EHeadType.Normal;
		}
		
		if (( sign != null )) 
		{
			__temp_me62.sign = sign;
		}
		 else 
		{
			__temp_me62.sign = nx3.ESign.None;
		}
		
		if (( tie != null )) 
		{
			__temp_me62.tie = tie;
		}
		 else 
		{
			__temp_me62.tie = null;
		}
		
		if (( tieTo != null )) 
		{
			__temp_me62.tieTo = tieTo;
		}
		 else 
		{
			__temp_me62.tieTo = null;
		}
		
		__temp_me62.level = ((int) (haxe.lang.Runtime.toInt(__temp_level61)) );
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.NHead(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.NHead(((nx3.EHeadType) (arr.__get(0)) ), ((java.lang.Object) (arr.__get(1)) ), ((nx3.ESign) (arr.__get(2)) ), ((nx3.ETie) (arr.__get(3)) ), ((nx3.ETie) (arr.__get(4)) ));
	}
	
	
	public  int level;
	
	public  nx3.EHeadType type;
	
	public  nx3.ESign sign;
	
	public  nx3.ETie tie;
	
	public  nx3.ETie tieTo;
	
	public  nx3.NNote nnote;
	
	@Override public   java.lang.String toString()
	{
		java.lang.String str = ( "" + this.level );
		if (( this.type != nx3.EHeadType.Normal )) 
		{
			str += ( " " + haxe.root.Type.enumConstructor(this.type) );
		}
		 else 
		{
			str += "";
		}
		
		if (( this.sign != nx3.ESign.None )) 
		{
			str += ( " " + haxe.root.Type.enumConstructor(this.sign) );
		}
		 else 
		{
			str += "";
		}
		
		return ( ( "NHead(" + str ) + ")" );
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef410 = true;
			switch (field.hashCode())
			{
				case 102865796:
				{
					if (field.equals("level")) 
					{
						__temp_executeDef410 = false;
						this.level = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef410) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef411 = true;
			switch (field.hashCode())
			{
				case 104974688:
				{
					if (field.equals("nnote")) 
					{
						__temp_executeDef411 = false;
						this.nnote = ((nx3.NNote) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 102865796:
				{
					if (field.equals("level")) 
					{
						__temp_executeDef411 = false;
						this.level = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 110356267:
				{
					if (field.equals("tieTo")) 
					{
						__temp_executeDef411 = false;
						this.tieTo = ((nx3.ETie) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef411 = false;
						this.type = ((nx3.EHeadType) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 114832:
				{
					if (field.equals("tie")) 
					{
						__temp_executeDef411 = false;
						this.tie = ((nx3.ETie) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3530173:
				{
					if (field.equals("sign")) 
					{
						__temp_executeDef411 = false;
						this.sign = ((nx3.ESign) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef411) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef412 = true;
			switch (field.hashCode())
			{
				case -1776922004:
				{
					if (field.equals("toString")) 
					{
						__temp_executeDef412 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("toString"))) );
					}
					
					break;
				}
				
				
				case 102865796:
				{
					if (field.equals("level")) 
					{
						__temp_executeDef412 = false;
						return this.level;
					}
					
					break;
				}
				
				
				case 104974688:
				{
					if (field.equals("nnote")) 
					{
						__temp_executeDef412 = false;
						return this.nnote;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef412 = false;
						return this.type;
					}
					
					break;
				}
				
				
				case 110356267:
				{
					if (field.equals("tieTo")) 
					{
						__temp_executeDef412 = false;
						return this.tieTo;
					}
					
					break;
				}
				
				
				case 3530173:
				{
					if (field.equals("sign")) 
					{
						__temp_executeDef412 = false;
						return this.sign;
					}
					
					break;
				}
				
				
				case 114832:
				{
					if (field.equals("tie")) 
					{
						__temp_executeDef412 = false;
						return this.tie;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef412) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef413 = true;
			switch (field.hashCode())
			{
				case 102865796:
				{
					if (field.equals("level")) 
					{
						__temp_executeDef413 = false;
						return ((double) (this.level) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef413) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef414 = true;
			switch (field.hashCode())
			{
				case -1776922004:
				{
					if (field.equals("toString")) 
					{
						__temp_executeDef414 = false;
						return this.toString();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef414) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("nnote");
		baseArr.push("tieTo");
		baseArr.push("tie");
		baseArr.push("sign");
		baseArr.push("type");
		baseArr.push("level");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


